tests/host_reference/domain_wall_dslash_reference.cpp

```diff
@@ -31,55 +32,93 @@ using namespace quda;
  * @param[in] parity The parity of the dslash (0 for even, 1 for odd).
  * @param[in] dagger Whether to apply the original or the Hermitian conjugate operator
  */
-template <QudaPCType type, typename real_t>
-void dslashReference_4d(real_t *out, const real_t *const *gauge, real_t const *const *ghostGauge, const real_t *in,
-                        const real_t *const *fwdSpinor, const real_t *const *backSpinor, int parity, int dagger)
-{
+template <typename real_t, QudaPCType type> struct DslashReference4D {
+  void operator()(void *out_, const void *const *gauge_, const void *const *ghostGauge_, const void *in_,
+                  const void *const *fwdSpinor_, const void *const *backSpinor_, int parity, int dagger)
+  {
+    real_t *out = reinterpret_cast<real_t *>(out_);
+    const real_t *const *gauge = reinterpret_cast<const real_t *const *>(gauge_);
```

Comment:
Minor point here: you could mark the lhs for these lines as `auto` to make the lines a easier to read IMHO.

Reply:
https://github.com/lattice/quda/commit/0d34df8423a88838d511df8120cd143a257c552f